src/os/__osDequeueThread.c

```diff
@@ -1,23 +1,24 @@
 #include "libultra_internal.h"
 
-NO_REORDER OSThread* __osThreadTail = NULL;
-NO_REORDER u32 __osTest = -1;
-NO_REORDER OSThread* __osRunQueue = (OSThread*) &__osThreadTail;
-NO_REORDER OSThread* __osActiveQueue = (OSThread*) &__osThreadTail;
-OSThread* __osRunningThread = NULL;
-OSThread* __osFaultedThread = NULL;
+struct __osThreadTail __osThreadTail = { NULL, -1 };
```

Comment:
did you update from https://github.com/n64decomp/libreultra/blob/1aca5c13ca041cef86f8dc194b727361dad9c09b/src/os/thread.c ? and if yes why not rename it ?

Reply:
I used ultralib, I would leave the renaming for another PR since I just focused on fixing types

Reply:
ok ok

Reply:
This change regarding threadtail stuff is actually a reversion.

I don't really know which verison is better or not.

Reply:
It's on a struct so there's no reordering, and it still matches

Reply:
Does this work on GCC?

Reply:
It does boot on GCC, but it still has some problems unrelated to this